pilot/pkg/bootstrap/server.go

```diff
@@ -467,6 +468,11 @@ func (s *Server) initIstiodAdminServer(args *PilotArgs, wh *inject.Webhook) erro
 	// Debug Server.
 	s.XDSServer.InitDebug(s.monitoringMux, s.ServiceController(), args.ServerOptions.EnableProfiling, wh)
 
+	// Add debug handlers on readinessMux for backward compatibility.
+	if features.EnableDebugOnHTTPAddr && args.ServerOptions.MonitoringAddr != "" {
```

Comment:
Why do you check MonitoringAddr ? monitoringAddr=="" means to multiplex it on the single port.

Reply:
Ah, I see - it is added on the monitoring mux anyways, and can't be added 2x if they are muxed.

A bit confusing - can you add a comment - 'debug handlers are currently added on monitoring mux and readiness mux. If monitoring addr is empty, the mux is shared and we only add it once on the shared mux'